src/Common/CurrentThread.h

```diff
@@ -62,9 +62,9 @@ class CurrentThread
     static void updatePerformanceCountersIfNeeded();
 
     static ProfileEvents::Counters & getProfileEvents();
-    inline ALWAYS_INLINE static MemoryTracker * getMemoryTracker()
```

Comment:
__always_inline to my understanding is a different construct with a different semantics from C++'s `inline` specifier. we shouldn't delete any of them.

Reply:
Oh, I deleted too much here, it should have been

```cpp
ALWAYS_INLINE static MemoryTracker * getMemoryTracker()
```

(note that we can't delete both `inline` and `ALWAYS_INLINE` because in that case, the clang-tidy warning will trigger)

I just noticed that a bigger problem is that

```cpp
ALWAYS_INLINE static MemoryTracker * getMemoryTracker()
```

doesn't necessarily inline. At least in Godbolt, it is possible to get non-inlined code for such code: https://godbolt.org/z/Eefd74qdM

That's bad because it makes `readability-redundant-inline-specifier` incompatible with ALWAYS_INLINE. So let me restore the suppression for `readability-redundant-inline-specifier`, add a comment to .clang-tidy, and revert the places where we use ALWAYS_INLINE. The other places (the majority of this PR, i.e. no ALWAYS_INLINE) are okay to change and it also makes sense to do so - too many `inline`s tend to be annoying.